File: TicTacToeCore/src/main/java/core/Field.java
package core;

record Field(int fieldNo, int playerNo, int currentNominal) {

    public static Field createNewField(int fieldNo) {
        Validator.validateFiledNo(fieldNo);
        return new Field(fieldNo, 0, 0);
    }

    public static Field updateField(Field previuos, Player currentPlayer, int newNominal) {
        Validator.validateNominalAllowed(previuos.currentNominal, newNominal);
        return new Field(previuos.fieldNo(), currentPlayer.getNo(), newNominal);
    }

}

File: TicTacToeFrontend/src/model/Field.ts
export class Field {

  fieldNo: number;
  playerNo: number;
  currentNominal: number

  constructor(fieldNo: number, playerNo: number, currentNominal: number) {
    this.fieldNo = fieldNo;
    this.playerNo = playerNo;
    this.currentNominal = currentNominal;
  }
}
